sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala

```diff
@@ -168,17 +168,26 @@ final case class WrappedGpuMetric(sqlMetric: SQLMetric) extends GpuMetric {
   override def value: Long = sqlMetric.value
 }
 
-class CollectTimeIterator(
+/** A GPU metric class that just accumulates into a variable without implicit publishing. */
+final class LocalGpuMetric extends GpuMetric {
```

Comment:
nice, do you think it makes sense to have this be a pattern we follow everywhere?

Reply:
This was needed in the case where we want to call code that really wants a GpuMetric but we're not sure yet _which_ metric it needs to be.  In the case of probing the left and right inputs, we are counting collect time for the stream and build side, but we don't know yet which side is the stream side and which is the build side.  The code that does the collect expects a GpuMetric to update, yet we don't know which one is the right one to pass until we're done collecting one side.  This works around that case.

Not sure we have other cases that are similar where we want to call code that needs a GpuMetric but don't know yet which one to call.  Usually we either already know which metric to update or don't want to publish that data at all (i.e.: NoopMetric).

Reply:
Ah ok, right, I think publishing happens after the fact anyway, I was thinking this could be a performance improvement, but I don't think it is